src/instrumentsscene/view/instrumentlistmodel.cpp

```diff
@@ -521,9 +539,9 @@ void InstrumentListModel::setCurrentGroup(const QString& groupId)
     loadInstruments();
 }
 
-void InstrumentListModel::doSetCurrentGroup(const QString& groupId)
+void InstrumentListModel::doSetCurrentGroup(const QString& groupId, bool compareWithCurrentGroup)
 {
-    if (m_currentGroupId == groupId) {
+    if (m_currentGroupId == groupId && compareWithCurrentGroup) {
```

Comment:
This code looks weird. Please explain why you need it

Reply:
Well the reason why was because if you were to just call `doSetCurrentGroup()` without my modification, the function would just check whether the current and the next groups are same. However, there are cases where the groups can be the same, but the index would vary. Index means the position of the group in the list

So if you don't consider that, then this will happen:

https://user-images.githubusercontent.com/88136054/178957159-c5386e21-22e2-48e0-8161-cbfc3aa1f28d.mp4


But now that you mention it, I think I should just add a check to see if the index is also the same inside the function rather than doing it outside the function.